src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -315,6 +318,36 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
     return results;
   }
 
+  /**
+   * Similar to {@link IntegrationTestUtils#waitUntilMinKeyValueRecordsReceived(Properties, String, int)}, except for
+   * use with {@link TopologyTestDriver} tests. Because the test driver is synchronous, we don't need to poll for
+   * the expected number of records, and then hope that these are all the results from our test. Instead, we can
+   * just read out <em>all</em> the processing results, for use with deterministic validations.
+   *
+   * @param topic Topic to consume from
+   * @param topologyTestDriver The {@link TopologyTestDriver} to read the data records from
+   * @param keyDeserializer The {@link Deserializer} corresponding to the key type
+   * @param valueDeserializer  The {@link Deserializer} corresponding to the value type
+   * @param <K> Key type of the data records
+   * @param <V> Value type of the data records
+   * @return A {@link List} of {@link KeyValue} pairs of results from the output topic, in the order they were produced.
+   */
+  static <K, V> List<KeyValue<K, V>> drainStreamOutput(final String topic,
```

Comment:
FYI; This seems to be similar than TestOutputTopic what I was proposing to add kafka-streams-test-utils in KIP-456

Please, take a look:
https://cwiki.apache.org/confluence/display/KAFKA/KIP-456%3A+Helper+classes+to+make+it+simpler+to+write+test+logic+with+TopologyTestDriver

Reply:
Also related to this. Would it be clearer to make separate TopologyTestUtils.java class to add these methods to it instead of adding those to IntegrationTestUtils.java?

Reply:
I personally think it's ok to add it here. Keeps all test utils unified on one class. We also don't have separate classed for producer and consumer test utils.

Reply:
About the KIP. It the KIP is available, it makes sense to use the new feature. Note, that this PR is agains older branches and won't be able to use the KIP in any case.

Reply:
Hi, @jukkakarvanen ! I haven't gotten a chance to look at your KIP yet, but I really like it. I'd be happy to come back and use new test-util improvements to this repo once they're available. Really, this PR is what we should have done when we added TopologyTestDriver, for example.

For now, though, I think I'd agree with @mjsax and rather leave all the test utility methods in one "bucket" to help us discover them as we maintain these tests in the future. Since TopologyTestDriver tests are also "integration tests", it seems like the name "IntegrationTestUtils" is still appropriate.